### PR TITLE
FEM-2521: fix downloading audio-only assets

### PR DESCRIPTION
**Pull request by noamtamim**
Download of audio-only assets was broken in #35 - this change fixes that.

Breaking change:
- The enums `VideoCodec` and `AudioCodec` in `DTGSelectionOptions` have been merged to a single enum, `TrackCodec`

### CODE REVIEW

**Comment by noamtamim** [comment]
The PR changes without the "split to 3 files" commit that caused a lot of false diffs:
https://github.com/kaltura/playkit-ios-dtg/compare/develop..2aa74361f68a9551451a180c24dc91c350204e7c

File: Sources/HLSLocalizer.swift [hunk 1]
```diff
@@ -583,37 +344,30 @@ class HLSLocalizer {
...
+        if hasVideo {
+            for codec in options.fullVideoCodecPriority() {
+                if let codecStreams = mainStreams[codec], let first = codecStreams.first {
+                    return try VideoStream(streamInfo: first, mediaUrl: first.m3u8URL(), type: M3U8MediaPlaylistTypeVideo)
+                }
+            }
+        } else if hasAudio {
+            for codec in options.fullAudioCodecPriority() {
+                if let codecStreams = mainStreams[codec], let first = codecStreams.first {
+                    return try VideoStream(streamInfo: first, mediaUrl: first.m3u8URL(), type: M3U8MediaPlaylistTypeVideo)
```
Comment (x-NR-x) [inline Sources/HLSLocalizer.swift]: type should be: M3U8MediaPlaylistTypeAudio

File: Sources/HLSLocalizerUtils.swift [hunk 2]
```diff
@@ -0,0 +1,369 @@
...
+    
+    let sorted = streams.stableSorted(by: sortOrder)
+    
+    let filtered = sorted.filter( filter )
+    
+    if filtered.isEmpty {
+        if let s = sorted.last {
+            return [s]
+        } else {
+            return []
```
Comment (x-NR-x) [inline Sources/HLSLocalizerUtils.swift]: The else can be removed, will never get there.

Comment (noamtamim) [reply Sources/HLSLocalizerUtils.swift]: Fixed

File: Sources/HLSLocalizer.swift (hunk 1 above)
Comment (noamtamim) [reply Sources/HLSLocalizer.swift]: Fixed

**Review by x-NR-x — APPROVED** [review]

**Pull request merged**